src/k8s/pkg/k8s/client/cluster.go

```diff
@@ -66,6 +66,11 @@ func (c *k8sdClient) Bootstrap(ctx context.Context, bootstrapConfig apiv1.Bootst
 // ClusterStatus returns the current status of the cluster.
 func (c *k8sdClient) ClusterStatus(ctx context.Context, waitReady bool) (apiv1.ClusterStatus, error) {
 	var response apiv1.GetClusterStatusResponse
+
+	if !waitReady && !c.IsKubernetesAPIServerReady(ctx) {
```

Comment:
Please move this to the GET /cluster endpoint. You moved the check into the client implementation. So, it will still be executed on the client-side.

Reply:
Thanks!